code/Control/UserDefinedFormController.php

```diff
@@ -252,7 +252,9 @@ public function process($data, $form)
                             $upload->loadIntoFile($_FILES[$field->Name], $file, $foldername);
                         } catch (ValidationException $e) {
                             $validationResult = $e->getResult();
-                            $form->addErrorMessage($field->Name, $validationResult->message(), 'bad');
+                            foreach ($validationResult->getMessages() as $message) {
+                                $form->sessionMessage($message['message'], 'bad');
```

Comment:
One minor suggestion:

```suggestion
                                $form->sessionMessage($message['message'], ValidationResult::TYPE_ERROR);
```

You may need to add a namespace too